### PR TITLE
refactor: split runtime types

### LINKED ISSUE #211: Fix types resolution in IDE
Waiting for the module builder of the nuxt3 framework team.

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Refactor runtime types so they are bundled with `mkdist`. resolves #211.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -19,7 +51,7 @@ export default defineNuxtModule<StrapiOptions>({
     version: 'v4',
     cookie: {}
   },
-  setup (options: StrapiOptions, nuxt: Nuxt) {
+  setup (options: ModuleOptions, nuxt: Nuxt) {
```
Comment (pi0) [inline src/module.ts:54]: Tip: You don't need types here. They are already inferred by `defineNuxtModule`

Comment (benjamincanac) [reply src/module.ts:54]: Didn't know that, should we remove it then?

**Review by benjamincanac — APPROVED** [review]

**Comment by benjamincanac** [comment]
Thanks a lot @pi0 🙂

File: src/module.ts [hunk 2]
```diff
@@ -2,9 +2,41 @@ import defu from 'defu'
...
+  */
+  prefix?: string
+
+  /**
+   * Strapi Version
+   * @default 'v4'
+   * @type string
+   * @example 'v3'
+   */
+  version?: 'v4' | 'v4'
```
Comment (luke-z) [inline src/module.ts:29]: Hi @pi0

Shouldn't this be `'v4' | 'v3'`?

Comment (benjamincanac) [reply src/module.ts:29]: @luke-z Thanks for the heads up I missed this, fixed in 6ad2e38ec4434db1d0dd1342502322eebd845948.

Comment (luke-z) [reply src/module.ts:29]: Awesome :)

**Pull request merged**